mdpow/fep.py

```diff
@@ -583,20 +584,9 @@ def setup(self, **kwargs):
         kwargs.setdefault('qscript', qscripts)
 
         for component, lambdas in self.lambdas.items():
-            if self.method == "TI":
-                for l in lambdas:
-                    # set up gromacs job for each FEP window in TI
-                    params = self._setup(component, l, **kwargs)
-            elif self.method == "BAR":
-                xlambdas = [None] + list(lambdas) + [None]
-                for l in lambdas:
-                    foreign_lambdas = [xlambdas[xlambdas.index(l) - 1],
-                                       xlambdas[xlambdas.index(l) + 1]]
-                    # set up gromacs job for each FEP window in BAR
-                    params = self._setup(component, l,
-                                         foreign_lambdas=foreign_lambdas, **kwargs)
-            else:
-                raise ValueError("Unknown method {}".format(self.method))
+            for l in lambdas:
```

Comment:
It's should be sufficient to supply the full lambda vector along with the `init_lambda_state`. The iterations over lambda was originally in the code; lambdas would be stripped out of the lambda list and the remaining lambdas would be put into the `fep-lambdas` mdp option. Now each MDP will contain information on the full lambda vector, which outputs information more consistently.